compiler/rustc_const_eval/src/interpret/cast.rs

```diff
@@ -64,6 +64,12 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 self.write_immediate(*res, dest)?;
             }
 
+            CastKind::PtrToMetadata => {
```

Comment:
Is this covered by Miri tests? Not sure how well we are covering the `ptr::metadata` API surface.